test/requests/code_actions_formatting_test.rb

```diff
@@ -0,0 +1,173 @@
+
+  # "Disable X for this line"
+
+  def test_emoji
+    assert_fixtures_match(
+      "emoji",
+      "Lint/UselessAssignment",
+      "Disable Lint/UselessAssignment for this line",
+    )
+  end
+
+  def test_lambda_indentation
+    assert_fixtures_match(
+      "lambda_indentation",
+      "Layout/IndentationConsistency",
+      "Disable Layout/IndentationConsistency for this line",
+    )
+  end
+
+  def test_comment_on_comment
+    assert_fixtures_match(
+      "comment_on_comment",
+      "Layout/CommentIndentation",
+      "Disable Layout/CommentIndentation for this line",
+    )
+  end
+
+  def test_multiline_array
+    assert_fixtures_match(
+      "multiline_array",
+      "Layout/IndentationConsistency",
+      "Disable Layout/IndentationConsistency for this line",
+    )
+  end
+
+  def test_line_disable_existing
+    assert_fixtures_match(
+      "line_disable_existing",
+      "Lint/UselessAssignment",
+      "Disable Lint/UselessAssignment for this line",
+    )
+  end
+
+  def test_multiline_string
+    skip("unimplemented")
+    assert_fixtures_match(
+      "multiline_string",
+      "Lint/Void",
+      "Disable Lint/Void for this line",
+    )
+  end
+
+  # "Disable X for this entire file"
+
+  def test_file_disable_first
+    skip("Document#push_edits not matching vscode")
+    assert_fixtures_match(
+      "file_disable_first",
+      "Lint/UselessAssignment",
+      "Disable Lint/UselessAssignment for this entire file",
+    )
+  end
+
+  def test_file_disable_no_trailing
+    skip("Document#push_edits not matching vscode")
+    assert_fixtures_match(
+      "file_disable_no_trailing",
+      "Lint/UselessAssignment",
+      "Disable Lint/UselessAssignment for this entire file",
+    )
+  end
+
+  def test_file_disable_existing
+    assert_fixtures_match(
+      "file_disable_existing",
+      "Lint/UselessAssignment",
+      "Disable Lint/UselessAssignment for this entire file",
+    )
+  end
+
+  def test_file_disable_existing_no_trailing
+    assert_fixtures_match(
+      "file_disable_existing_no_trailing",
+      "Lint/UselessAssignment",
+      "Disable Lint/UselessAssignment for this entire file",
+    )
+  end
+
+  def test_file_disable_mismatch
+    skip("Document#push_edits not matching vscode")
+    assert_fixtures_match(
+      "file_disable_mismatch",
+      "Lint/UselessAssignment",
+      "Disable Lint/UselessAssignment for this entire file",
+    )
+  end
+
+  private
+
+  def assert_fixtures_match(name, diagnostic_code, code_action_title)
+    actual, expected = load_expectation(name)
+    assert_corrects_to_expected(
+      diagnostic_code,
+      code_action_title,
+      actual,
+      expected,
+    )
+  end
+
+  sig do
+    params(
+      diagnostic_code: String,
+      code_action_title: String,
+      source: String,
+      expected: String,
+    ).returns(T.untyped)
+  end
+  def assert_corrects_to_expected(diagnostic_code, code_action_title, source, expected)
```

Comment:
I think this would be simpler if we just applied all available code actions for a fixture and then verified that the `document.source` matches the expectation file.

We lose a bit of granularity in the testing, but I think that's fine.